imagersite/imager_images/views.py

```diff
@@ -1,3 +1,43 @@
-from django.shortcuts import render
+"""Views for imager_images."""
 
-# Create your views here.
+from django.shortcuts import render, redirect, get_object_or_404
+from .models import Album, Photo
+from django.contrib.auth.models import User
+
+
+def library_view(request, username=None):
+    """Define the library view."""
+    owner = False
+
+    if not username:
+        username = request.user.get_username()
+        owner = True
+        if username == '':
+            return redirect('home')
+    # import pdb; pdb.set_trace()
+    profile = get_object_or_404(User, username=username)
+    albums = Album.objects.filter(user__username=username)
+    photos = Photo.objects.filter(user__username=username)
+    public_photos = photos.filter(published='PUBLIC')
+    public_albums = albums.filter(published='PUBLIC')
+
+    if not owner:
+        photos = Photo.objects.filter(published='PUBLIC')
+        albums = Album.objects.filter(published='PUBLIC')
+
+    num_albums = len(albums)
+    num_photos = len(photos)
+    num_public_albums = len(public_albums)
+    num_public_photos = len(public_photos)
```

Comment:
You do not need to take the length of the these here. The `QuerySet` result has a `.count()`method that you can access in your template as `.count`.